modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1385,6 +1389,15 @@ class CV_EXPORTS_W_SIMPLE TextRecognitionModel : public Model
     CV_WRAP
     const std::string& getDecodeType() const;
 
+    /**
+     * @brief Set the decoding method options, currently only for 'CTC-prefix-beam-search' usage
+     * @param[in] beam Beam size for search
+     * @param[in] vocPrune Parameter to optimize big vocabulary search,
+     * only take top @p vocPrune tokens in each search step, @p vocPrune <= 0 stands for disable this prune.
+     */
+    CV_WRAP
+    TextRecognitionModel& setDecodeOpts(int beam, int vocPrune = 0);
```

Comment:
Perhaps, it makes sense to name this as `setDecodeOptsCTCPrefixBeamSearch` to avoid confusions in the future.

Reply:
Fixed